src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ArrayModelBinder.cs

```diff
@@ -9,21 +9,19 @@ namespace Microsoft.AspNet.Mvc.ModelBinding
 {
     public class ArrayModelBinder<TElement> : CollectionModelBinder<TElement>
     {
-        public override Task<bool> BindModelAsync(ModelBindingContext bindingContext)
+        public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
         {
             if (bindingContext.ModelMetadata.IsReadOnly)
             {
-                return Task.FromResult(false);
+                return Task.FromResult<ModelBindingResult>(null);
```

Comment:
ok, Ill repeat what @rynowak says
>null is not failed. null is null - it specifies the lack of a result.

:smile: lets fix it if we really think adding a state would be more productive ( I feel it is more confusing than anything, as there is already a IsModelSet flag.), a success/failure state will increase the matrix to 2x2 with the fourth state not adding any value.